js/plugins/pinecone/src/index.ts

```diff
@@ -185,7 +187,7 @@ export function configurePineconeIndexer<
   textKey?: string;
   embedder: EmbedderArgument<EmbedderCustomOptions>;
   embedderOptions?: z.infer<EmbedderCustomOptions>;
-}) {
+}): IndexerAction<z.ZodOptional<typeof PineconeIndexerOptionsSchema>> {
```

Comment:
This seemed a little funky to me. Not sure if this is 💯 .

It looks like we're possibly double-wrapping in an Optional? https://github.com/firebase/genkit/blob/main/js/ai/src/embedder.ts#L74-L78